util/headerreader/header_reader.go

```diff
@@ -88,13 +91,19 @@ var TestConfig = Config{
 }
 
 func New(client arbutil.L1Interface, config ConfigFetcher) *HeaderReader {
+	isParentChainArbitrum := false
+	_, err := precompilesgen.NewArbSys(types.ArbSysAddress, client)
+	if err == nil {
+		isParentChainArbitrum = true
```

Comment:
This is confusing and it got me too, but this will only ever be an error if the ArbSys ABI is incorrect. It doesn't check whether the contract actually exists.

Reply:
ah, updated to check the block number in that case.